public/main.js

```diff
@@ -3,38 +3,111 @@ import { OrbitControls } from 'three/addons/controls/OrbitControls.js';
 import { GLTFLoader } from 'three/addons/loaders/GLTFLoader.js';
 
 const scene = new THREE.Scene();
-const camera = new THREE.PerspectiveCamera( 75, window.innerWidth / window.innerHeight, 0.1, 1000 );
+const camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, 0.1, 1000);
 const renderer = new THREE.WebGLRenderer();
-const controls = new OrbitControls( camera, renderer.domElement );
+const controls = new OrbitControls(camera, renderer.domElement);
 controls.enablePan = false;
+controls.minDistance = 0.75; 
 const gltf_loader = new GLTFLoader();
 
 const raycaster = new THREE.Raycaster();
 const mouse = new THREE.Vector2();
+const textureLoader = new THREE.TextureLoader();
+
+// Get current month and corresponding image name
+const date = new Date();
+const month = date.getMonth();
+const monthNames = [
+    "january", "february", "march", "april", "may", "june",
+    "july", "august", "september", "october", "november", "december"
+];
+const imageName = `meshes/earth/earth_${monthNames[month]}.jpg`;
+console.log("Selected image for the current month:", imageName);
+
+const fakeSatelliteData = [
+    { id: 'satellite_1', position: { x: 0.5, y: -1, z: 0.1 } },
+    { id: 'satellite_2', position: { x: -2, y: 0.2, z: 0.2 } },
+    { id: 'satellite_3', position: { x: 0.3, y: 0.3, z: -1.5 } },
+];
+
+const mockSatelliteData = {
+    'satellite_1': {
+        name: 'Satellite Alpha',
+        launchDate: '2020-01-01',
+    },
+    'satellite_2': {
+        name: 'Satellite Beta',
+        launchDate: '2021-05-15',
+    },
+    'satellite_3': {
+        name: 'Satellite C',
+        launchDate: '2022-12-15',
+    },
+};
+
+function fetchSatelliteData() {
```

Comment:
Add a fallback/error-handling to these two functions (in case the networking goes wrong or something else)